pkg/cnb/image_rebaser.go

```diff
@@ -21,16 +21,15 @@ type ImageRebaser struct {
 }
 
 func (f *ImageRebaser) Rebase(build *v1alpha1.Build, ctx context.Context) (BuiltImage, error) {
-	builderImage, err := f.RemoteImageFactory.NewRemote(build.Spec.Builder.Image, registry.SecretRef{
+	builderImage, err := f.RemoteImageFactory.newRemote(build.Spec.Builder.Image, build.Spec.Builder.Image, registry.SecretRef{
 		Namespace:        build.Namespace,
 		ImagePullSecrets: build.Spec.Builder.ImagePullSecrets,
-		ServiceAccount:   build.Spec.ServiceAccount,
```

Comment:
Will there be any case where ImagePullSecrets are empty and kpack as to rely on the service account to get credentials for a private builder?

Reply:
No. kpack should not use the build's service account to fetch the builder.